bin/cron/update_pledge_map

```diff
@@ -26,34 +26,60 @@ CENSUS_FORM_KINDS = "'Census2017','HocCensus2017'".freeze
 file = Tempfile.new(['data', '.csv'])
 begin
   CSV.open(file.path, 'w') do |csv|
+    DASHBOARD_DB[:census_submissions].
+      select_append(:census_submissions__id___census_submissions_id).
+      join(:census_submissions_school_infos, census_submission_id: :id).
+      join(:school_infos, id: :school_info_id).
+      join(:schools, id: :school_id).
+      paged_each(rows_per_fetch: 10_000) do |submission|
+      pledged = submission[:pledged]
+      teaches_yes = ["SOME", "ALL"]
+      teaches_cs = teaches_yes.include?(submission[:how_many_10_hours]) || teaches_yes.include?(submission[:how_many_20_hours])
+
+      # Only add to the map if pledge checkbox is checked or the school teaches some form of CS.
+      next unless pledged || teaches_cs
+
+      type = pledged ? "pledged" : "teaches_cs"
+
+      next unless submission[:school_id]
+      row_id = "CENSUS_SUBMISSIONS-#{submission[:census_submissions_id]}"
+      row = [row_id, submission[:name], submission[:zip], submission[:city], type]
+      csv << row
+    end
+
     DB_READONLY[:forms].where("kind in (#{CENSUS_FORM_KINDS})").paged_each(rows_per_fetch: 10_000) do |form|
-      data = JSON.parse(form[:data])
+      # Skip any rows that were migrated to CENSUS_SUBMISSIONS since those were already processed above.
+      form_map = DASHBOARD_DB[:census_submission_form_maps].where(form_id: form[:id])
+      if form_map.empty?
```

Comment:
Ruby style tries to minimize long scoped `if` and other blocks. This would be better as `next if form_map.empty?` then you probably can leave the rest of this method unchanged?

Reply:
Good point. I'll make this change.